### PR TITLE
More cppcheck fixes, fixes #22702

### LINKED ISSUE #22702: Cppcheck errors in 3.1
Cppcheck version 1.76.1
Godot version 3.1 rev caa141a1ac630144e441f89689baa1a74bb3bae5

I fixed the easy ones in the PR below.

All remaining cppcheck errors in non third party code:

[core/io/logger.cpp:196]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:200]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:202]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:250]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:251]: (error) va_list 'list_copy' used before va_start() was called.
[core/io/logger.cpp:252]: (error) va_list 'list_copy' used before va_start() was called.
[core/os/memory.cpp:101]: (error) Memory leak: mem
[editor/progress_dialog.cpp:60]: (error) Uninitialized struct member: t.vb
[editor/progress_dialog.cpp:60]: (error) Uninitialized struct member: t.state
[editor/progress_dialog.cpp:121]: (error) Uninitialized variable: no_updates
[editor/progress_dialog.cpp:183]: (error) Uninitialized struct member: t.hb
[main/input_default.cpp:407]: (error) Uninitialized struct member: action.strength
[main/input_default.cpp:541]: (error) Uninitialized struct member: action.strength
[main/input_default.cpp:552]: (error) Uninitialized struct member: action.strength
[platform/iphone/os_iphone.cpp:566]: (error) Memory leak: new_ptr
[servers/physics/body_pair_sw.cpp:130]: (error) Uninitialized struct member: contact.local_shape
[servers/physics/body_pair_sw.cpp:136]: (error) Uninitialized struct member: contact.local_shape
[servers/physics/body_pair_sw.cpp:130]: (error) Uninitialized struct member: contact.collider_shape
[servers/physics/body_pair_sw.cpp:136]: (error) Uninitialized struct member: contact.collider_shape
[servers/physics/body_pair_sw.cpp:130]: (error) Uninitialized struct member: contact.active
[servers/physics/body_pair_sw.cpp:136]: (error) Uninitialized struct member: contact.active
[servers/visual/shader_language.cpp:2527]: (error) Uninitialized struct member: pos.char_idx
[servers/visual/shader_language.cpp:2527]: (error) Uninitialized struct member: pos.tk_line

Probably false positives:
[drivers/gles2/shader_gles2.h:240]: (error) Array 'values[3]' accessed at index 3, which is out of bounds.
[drivers/gles2/shader_gles2.h:256]: (error) Array 'values[3]' accessed at index 3, which is out of bounds.
[drivers/gles2/shader_gles2.h:272]: (error) Array 'values[3]' accessed at index 3, which is out of bounds.

### PR DESCRIPTION
**Pull request by swenner**
Fixes #22702

### CODE REVIEW

File: servers/visual/shader_language.cpp [hunk 1]
```diff
@@ -2498,7 +2498,7 @@ bool ShaderLanguage::_get_completable_identifier(BlockNode *p_block, CompletionT
 
 	identifier = StringName();
 
-	TkPos pos;
+	TkPos pos = {0, 0};
```
Comment (akien-mga) [inline servers/visual/shader_language.cpp]: `clang-format` requires this style here: https://travis-ci.org/godotengine/godot/jobs/445889428

Comment (swenner) [reply servers/visual/shader_language.cpp]: I'm a slow learner. Sorry for that :)

**Comment by akien-mga** [comment]
Could you squash the commits?

**Comment by swenner** [comment]
Squashed.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**